crates/networking/p2p/sync.rs

```diff
@@ -0,0 +1,137 @@
+use std::sync::Arc;
+
+use ethrex_blockchain::error::ChainError;
+use ethrex_core::{
+    types::{Block, BlockHash, BlockHeader},
+    H256,
+};
+use ethrex_storage::Store;
+use tokio::{sync::Mutex, time::Instant};
+use tracing::{debug, info, warn};
+
+use crate::kademlia::KademliaTable;
+
+/// Manager in charge the sync process
+/// Only performs full-sync but will also be in charge of snap-sync in the future
+#[derive(Debug)]
+pub struct SyncManager {
+    // true: syncmode = snap, false = syncmode = full
```

Comment:
nit: wouldn't it be clearer to have an enum instead of a bool?

Reply:
I actually find an enum fitting for this situation, having a bool can lead to wonder what false/true means (plus the enum has type checking which warns when you're not using some varint), but this is not something critical.